test/src/main/java/org/apache/accumulo/test/ScanServerConcurrentTabletScanIT.java

```diff
@@ -0,0 +1,220 @@
+
+    @Override
+    public void configureMiniCluster(MiniAccumuloConfigImpl cfg,
+        org.apache.hadoop.conf.Configuration coreSite) {
+      cfg.setNumScanServers(1);
+      cfg.setProperty(Property.TSERV_SESSION_MAXIDLE, "3s");
+      cfg.setProperty(Property.SSERV_MINTHREADS, "4");
+    }
+  }
+
+  @BeforeAll
+  public static void start() throws Exception {
+    ScanServerConcurrentTabletScanITConfiguration c =
+        new ScanServerConcurrentTabletScanITConfiguration();
+    SharedMiniClusterBase.startMiniClusterWithConfig(c);
+  }
+
+  @AfterAll
+  public static void stop() throws Exception {
+    SharedMiniClusterBase.stopMiniCluster();
+  }
+
+  private void startScanServer(boolean cacheEnabled)
+      throws IOException, KeeperException, InterruptedException {
+
+    String zooRoot = getCluster().getServerContext().getZooKeeperRoot();
+    ZooReaderWriter zrw = getCluster().getServerContext().getZooReaderWriter();
+    String scanServerRoot = zooRoot + Constants.ZSSERVERS;
+
+    SharedMiniClusterBase.getCluster().getClusterControl().stop(ServerType.SCAN_SERVER);
+
+    Map<String,String> overrides = new HashMap<>();
+    overrides.put(Property.SSERV_CACHED_TABLET_METADATA_EXPIRATION.getKey(),
+        cacheEnabled ? "300m" : "0m");
+    SharedMiniClusterBase.getCluster().getClusterControl().start(ServerType.SCAN_SERVER, overrides,
+        1);
+    while (zrw.getChildren(scanServerRoot).size() == 0) {
+      Thread.sleep(500);
+    }
+
+  }
+
+  @Test
+  public void testScanSameTabletDifferentDataTabletMetadataCacheEnabled() throws Exception {
+
+    startScanServer(true);
+
+    Properties clientProperties = getClientProps();
+    clientProperties.put(ClientProperty.SCANNER_BATCH_SIZE.getKey(), "100");
+
+    try (AccumuloClient client = Accumulo.newClient().from(clientProperties).build()) {
+      String tableName = getUniqueNames(1)[0];
+
+      client.tableOperations().create(tableName);
+
+      // Load 1000 k/v
+      ReadWriteIT.ingest(client, 10, 100, 50, 0, "COLA", tableName);
+      client.tableOperations().flush(tableName, null, null, true);
+
+      Scanner scanner1 = client.createScanner(tableName, Authorizations.EMPTY);
+      scanner1.setRange(new Range());
+      scanner1.setBatchSize(100);
+      scanner1.setReadaheadThreshold(0);
+      scanner1.setConsistencyLevel(ConsistencyLevel.EVENTUAL);
+
+      // iter1 should read 1000 k/v
+      Iterator<Entry<Key,Value>> iter1 = scanner1.iterator();
+
+      // Partially read the data and then start a 2nd scan
+      int count1 = 0;
+      while (iter1.hasNext() && count1 < 10) {
+        iter1.next();
+        count1++;
+      }
+
+      // Load another 100 k/v
+      ReadWriteIT.ingest(client, 10, 10, 50, 0, "COLB", tableName);
+      client.tableOperations().flush(tableName, null, null, true);
+
+      // iter2 should read 1000 k/v because the tablet metadata is cached.
+      Iterator<Entry<Key,Value>> iter2 = scanner1.iterator();
+
+      while (iter1.hasNext()) {
```

Comment:
Would it be worth it if we combine these two while loops and actually alternate between iter1 and iter2 to ensure everything is running concurrently?